phpBB/phpbb/db/driver/driver.php

```diff
@@ -793,6 +804,130 @@ function sql_build_query($query, $array)
 		return $sql;
 	}
 
+
+	protected function _process_boolean_tree_first($operations_ary)
+	{
+		// In cases where an array exists but there is no head condition,
+		// it should be because there's only 1 WHERE clause. This seems the best way to deal with it.
+		if ($operations_ary[0] !== 'AND' &&
+			$operations_ary[0] !== 'OR')
```

Comment:
Although it could. Part of the idea is to allow creating other connecting logic later if it makes sense.
Regardless, if you insist, I'll change it. I don't have any pressuring will to keep it like that. It was just an idea.